src/device-state/current-state.ts

```diff
@@ -108,25 +108,25 @@ const getStateDiff = (): DeviceStatus => {
 	}
 
 	const diff = {
-		local: _(stateForReport.local)
-			.omitBy((val, key: keyof DeviceStatus['local']) =>
-				_.isEqual(lastReportedLocal[key], val),
-			)
-			.omit(INTERNAL_STATE_KEYS)
-			.value(),
-		dependent: _(stateForReport.dependent)
-			.omitBy((val, key: keyof DeviceStatus['dependent']) =>
-				_.isEqual(lastReportedDependent[key], val),
-			)
-			.omit(INTERNAL_STATE_KEYS)
-			.value(),
+		local: _.omitBy(
+			stateForReport.local,
+			(val, key: keyof NonNullable<DeviceStatus['local']>) =>
```

Comment:
Without the `NonNullable` this was returning a type of `never` since undefined cannot have a `keyof` - it was working because apparently indexing with `never` just works?